src/modules/i18n/containers/__tests__/I18nProviderContainer.spec.js

```diff
@@ -2,40 +2,34 @@
 
 import configureMockStore from 'redux-mock-store'
 import { Provider } from 'react-redux'
-import { shallow } from 'enzyme'
 import I18nProviderContainer from '../I18nProviderContainer'
-import React from 'react'
-import I18nProvider from '../../components/I18nProvider'
-import AsyncStorage from '@react-native-community/async-storage'
+import * as React from 'react'
+import waitForExpect from 'wait-for-expect'
+import { render } from '@testing-library/react-native'
 
-const mockStore = configureMockStore()
+jest.mock('@react-native-community/async-storage')
+jest.mock('../../../i18n/LanguageDetector')
 
-const mockAsyncStorage: AsyncStorage = require('@react-native-community/async-storage/jest/async-storage-mock').default
-jest.mock('@react-native-community/async-storage', () => mockAsyncStorage)
+const mockStore = configureMockStore()
 
 describe('I18nProviderContainer', () => {
-  it('should pass the right props', () => {
+  it('should pass the right props', async () => {
     const store = mockStore({})
-    store.dispatch = jest.fn()
-    const wrapper = shallow(
+    render(
       <Provider store={store}>
         <I18nProviderContainer>
           <React.Fragment />
         </I18nProviderContainer>
       </Provider>
     )
 
-    const props = wrapper.dive().find(I18nProvider).props()
-    expect(props.children).toEqual(<React.Fragment />)
-
-    const setContentLanguage = props.setContentLanguage
-    setContentLanguage('de')
-    expect(store.dispatch).toHaveBeenCalledTimes(1)
-    expect(store.dispatch).toHaveBeenCalledWith({
-      type: 'SET_CONTENT_LANGUAGE',
-      params: {
-        contentLanguage: 'de'
-      }
+    await waitForExpect(() => {
+      expect(store.getActions()).toEqual([{
+        type: 'SET_CONTENT_LANGUAGE',
+        params: {
+          contentLanguage: 'en'
+        }
```

Comment:
You don't expect that the right props are passed (which is what the title of the test says)

Reply:
I added the unit test